programmers-coding-test/wonji/implement/게임개발.js

```diff
@@ -0,0 +1,72 @@
+const rl = require("readline-sync");
+const n = parseInt(rl.question("세로크기 N입력(3이상).\n>>"));
+const m = parseInt(rl.question("가로크기 N입력(50이하).\n>>"));
+
+let x = parseInt(rl.question("현재 캐릭터의 x좌표 입력.\n>>"));
+let y = parseInt(rl.question("현재 캐릭터의 y좌표 입력.\n>>"));
+let direction = parseInt(rl.question("현재 캐릭터의 위치방향 입력(0:북, 1:동, 2:남, 3:서(숫자로 입력하세요)).\n>>"));
+
+let map = 0;
+let mapArray = [];
+let mapSplit = 0;
+
+for (let i = 0; i < n; i++){
+  map = rl.question("지도를 작성하시오(0:육지, 1:바다).\n>>");
+  mapSplit = map.split("");
+  mapArray.push(mapSplit);
+}
```

Comment:
array와 split 같은 변수명은 최대한 피하는 것이 좋음. 복수형태일 경우는 뒤에 “s”를 붙여서 사용하자.
```suggestion
const map = [];
for (let i = 0; i < n; i++){
  const vector = rl.question("지도를 작성하시오(0:육지, 1:바다).\n>>").split(“ “); // 공백을 두고 입력해야함. 벡터는 1차원 행렬을 의미함.
  map.push(vector);
}
```